src/runtime/runtime_atsamd21.go

```diff
@@ -302,8 +305,18 @@ func timerSleep(ticks uint32) {
 	// enable IRQ for CMP0 compare
 	sam.RTC_MODE0.INTENSET.SetBits(sam.RTC_MODE0_INTENSET_CMP0)
 
-	for timerWakeup.Get() == 0 {
-		arm.Asm("wfi")
+wait:
+	arm.Asm("wfi")
```

Comment:
Huh, interesting. There is a subtle difference between `wfe` and `wfi`. I think it's best to ~always use `wfe`, as it avoids a race condition if the interrupt that you're waiting for happens right before the `wfi` instruction (the `wfi` would wait until the next interrupt, while `wfe` would be a `nop` in that case).

But it's somewhat unrelated as this code was already present.